src/Content/Text/BBCode.php

```diff
@@ -169,137 +169,54 @@ public static function getAttachmentData($body)
 
 		$data['text'] = trim($match[1]);
 
-		$type = '';
-		preg_match("/type='(.*?)'/ism", $attributes, $matches);
-		if (!empty($matches[1])) {
-			$type = strtolower($matches[1]);
-		}
-
-		preg_match('/type="(.*?)"/ism', $attributes, $matches);
-		if (!empty($matches[1])) {
-			$type = strtolower($matches[1]);
-		}
-
-		if ($type == '') {
-			return [];
+		foreach (['type', 'url', 'title', 'image', 'preview', 'publisher_name', 'publisher_url', 'author_name', 'author_url'] as $field) {
+			$value = '';
+			preg_match("/" . $field . "='(.*?)'/ism", $attributes, $matches);
+			if (!empty($matches[1])) {
+				$value = $matches[1];
+			}
+	
+			preg_match('/' . $field . '="(.*?)"/ism', $attributes, $matches);
+			if (!empty($matches[1])) {
+				$value = $matches[1];
+			}
+	
```

Comment:
Regular expression doctor here, these two can be combined in one:
```suggestion
			preg_match('/' . preg_quote($field, '/') . '=("|\')(.*?)\1/ism', $attributes, $matches);
			$value = $matches[2] ?? '';
```
The `\1` is a backreference to the first capturing group, here it is `("|')` (escape slash is added because of the single-quoted string). This enables us to look for quote pairs depending on the first found. It also relegates the value capturing group to be the second group and that's why we are testing on `$matches[2]` now.

The `preg_quote` isn't strictly necessary because we provide the whole list of field names, but it's still good practice because we don't cover these with unit tests.

Reply:
Okay, changed and locally tested.